src/components/InvestmentCalculator.tsx

```diff
@@ -0,0 +1,383 @@
+      startDateInUnix,
+      endDateInUnix
+    );
+    const priceAndDateForEachInterval =
+      getPriceAndDateForEachInvterval(historyData);
+    const actualPricesArray = priceAndDateForEachInterval.map(
+      (item) => item[1]
+    );
+    if (valueCostIsSelected) {
+      const returnFromVACalculation = calculateNetInvestmentVA(
+        parseInt(initialInvestment),
+        parseInt(targetGrowthRate),
+        actualPricesArray
+      );
+      setCalculationResult(returnFromVACalculation);
+    } else {
+      const returnFromDACalculation = calculateNetInvestmentDA(
+        parseInt(initialInvestment),
+        parseInt(investmentAdded),
+        actualPricesArray
+      );
+      setCalculationResult(returnFromDACalculation);
+    }
+  };
+
+  return (
+    <div className="text-skin-calculator-general-text-color">
+      <div
+        className="flex justify-center items-center w-52 h-10 mr-2 rounded-md mb-3 bg-skin-portfolio-item-buttons-background-color text-skin-portfolio-item-buttons-text-color cursor-pointer"
+        onClick={togglePopup}
+      >
+        Investment Calculator
+      </div>
+      {showPopup && (
+        <div className="fixed top-0 left-0 w-full h-full bg-black bg-opacity-50 flex justify-center items-center z-50 backdrop-filter backdrop-blur-md  font-space-grotesk">
+          <div
+            className={`fixed top-1/2 right-1/2 transform translate-x-1/2 -translate-y-1/2 w-[90%] md:w-[768px] lg:w-[820px] h-[620px] md:h-[768px] bg-skin-app ${
+              darkMode ? "" : "theme-light"
+            } rounded-lg border-gray-300 shadow-md py-10 md:pt-6 md:pb-10 px-6 md:px-10 bg-skin-calculator-pop-up-background-color`}
+          >
+            <div className="flex h-[7%] md:h-[12%] justify-between items-center">
+              <div className="font-space-grotesk text-lg font-semibold">
+                Investment Calculator
+              </div>
+              <svg
+                viewBox="0 0 24 24"
+                xmlns="http://www.w3.org/2000/svg"
+                fill={
+                  darkMode ? "rgba(255 ,255 ,255 ,1)" : "rgba(66, 66, 134, 1)"
+                }
+                className="w-8 h-8 cursor-pointer"
+                onClick={togglePopup}
+              >
+                <g id="SVGRepo_bgCarrier" stroke-width="0"></g>
+                <g
+                  id="SVGRepo_tracerCarrier"
+                  stroke-linecap="round"
+                  stroke-linejoin="round"
+                ></g>
+                <g id="SVGRepo_iconCarrier">
+                  {" "}
+                  <rect x="0" fill="none" width="24" height="24"></rect>{" "}
+                  <g>
+                    {" "}
+                    <path d="M19.1 4.9C15.2 1 8.8 1 4.9 4.9S1 15.2 4.9 19.1s10.2 3.9 14.1 0 4-10.3.1-14.2zm-4.3 11.3L12 13.4l-2.8 2.8-1.4-1.4 2.8-2.8-2.8-2.8 1.4-1.4 2.8 2.8 2.8-2.8 1.4 1.4-2.8 2.8 2.8 2.8-1.4 1.4z"></path>{" "}
+                  </g>{" "}
+                </g>
+              </svg>
```

Comment:
put the svh in a component so i can tell what this is and not polute the jsx with markup